src/com/amazon/ion/impl/bin/IonManagedBinaryWriter.java

```diff
@@ -725,414 +124,279 @@ public boolean isFieldNameSet()
+                            symbols.writeInt(st.getVersion());
+                            symbols.setFieldNameSymbol(systemSymbol(MAX_ID_SID));
+                            symbols.writeInt(st.getMaxId());
+                            symbols.stepOut();
+                        }
+                        symbols.stepOut();
+                    }
+                }
+                if (lstIndex < maxId) {
+                    //if our index of declared symbols is less than our context's maxId index then we have new local symbols.
+                    symbols.setFieldNameSymbol(systemSymbol(SYMBOLS_SID));
+                    symbols.stepIn(LIST);
+                    for (int i = this.lstIndex + 1; i <= maxId; i++) {
+                        symbols.writeString(this.lst.findKnownSymbol(i));
+                    }
+                    lstIndex = maxId;
+                    symbols.stepOut();
+                }
+                symbols.stepOut();
+                flushed = true;
+                newSymbols = false;
+            }
+            symbols.flush();
+            user.flush();
+        }
+    }
+
+    public void finish() throws IOException {
+        //resets the symbol context of the writer and finishes serializing payload.
+        //this allows for user to continue writing using the writer from a fresh symbol context as they see fit.
+        if (getDepth() != 0) throw new IllegalStateException("IonWriter.finish() can only be called at top-level.");
+        flush();
+        lstWriter = new _Private_LSTWriter(fallbackImports, new ArrayList<String>(), catalog);
+        lst = lstWriter.getSymbolTable();
+        lstIndex = lst.getImportedMaxId();
+        flushed = false;
+    }
+
+    public void close() throws IOException {
+        if (!closed) {
+            closed = true;
+            try {
+                flush();
+            } catch (IllegalStateException e) {
+            } finally {
+                try {
+                    symbols.close();
+                } finally {
+                    user.close();
+                }
+            }
         }
-
-        // make sure that until the local symbol state changes we no-op the table closing routine
-        symbolState = SymbolState.LOCAL_SYMBOLS_FLUSHED;
-        // push the data out
-        symbols.finish();
-        user.finish();
     }
 
-    public void finish() throws IOException
-    {
-        if (getDepth() != 0)
-        {
-            throw new IllegalStateException("IonWriter.finish() can only be called at top-level.");
-        }
-        unsafeFlush();
-        // Reset local symbols
-        // TODO be more configurable with respect to local symbol table caching
-        locals.clear();
-        localsLocked = false;
-        symbolState = SymbolState.SYSTEM_SYMBOLS;
-        imports = bootstrapImports;
+    public _Private_IonRawWriter getRawWriter() {
+        return user;
```

Comment:
Should this be `user` or `currentWriter`?

Reply:
user writer, surfacing the LST factory "writer" would be bad I think.